tests/unit/check-current-route-test.js

```diff
@@ -3,14 +3,152 @@ import { module, test } from 'qunit';
+      },
+      {
+        currentRoute: {
+          url: '/posts/comments/show',
+          name: 'posts.comments.show',
+        },
+        routeValue: {
+          url: '/posts/comments/show',
+          name: 'comments.show',
+        },
+        message: 'child resource and route matches correctly',
+      },
+      {
+        currentRoute: {
+          url: '/comments',
+          name: 'comments',
+        },
+        routeValue: {
+          url: '/comments',
+          name: 'comments',
+        },
+        message: 'route w/ resetNamespace matches correctly',
+      },
+      {
+        currentRoute: {
+          url: '/company/department/employees',
+          name: 'company.department.employees',
+        },
+        routeValue: {
+          url: '/company/department/employees',
+          name: 'employees',
+        },
+        message: 'nested route w/ resetNamespace matches correctly',
+      }
+    ];
 
-    assert.ok(checkCurrentRoute('posts.show', 'posts'), 'resource matches correctly');
-    assert.ok(!checkCurrentRoute('posts.show', 'comments'), 'resource matches correctly');
-    assert.ok(checkCurrentRoute('posts.comments.show', 'posts'), 'parent resource of nested resources matches correctly');
-    assert.ok(checkCurrentRoute('comments.show', 'comments.show'), 'exact resource and route matches correctly');
-    assert.ok(checkCurrentRoute('posts.comments.show', 'comments.show'), 'child resource and route matches correctly');
+    testSet.forEach(({ currentRoute, routeValue, message }) => {
+      assert.ok(checkCurrentRoute(currentRoute, routeValue), message);
+    });
+  });
+
+  test('does not match incorrect routes', function(assert) {
+    const testSet = [
+      {
+        currentRoute: {
+          url: '/posts/show',
+          name: 'posts.show',
+        },
+        routeValue: {
+          url: '/posts',
+          name: 'index',
+        },
+        message: 'resource match fails even when route name same as resource name',
+      },
+      {
+        currentRoute: {
+          url: '/posts/show',
+          name: 'posts.show',
+        },
+        routeValue: {
+          url: '/comments/show',
+          name: 'comments',
+        },
+        message: 'resource match fails when current route does not match url',
+      },
+      {
+        currentRoute: {
+          url: '/posts/show/comments',
+          name: 'posts.show.comments',
+        },
+        routeValue: {
+          url: '/users/comments',
+          name: 'comments',
+        },
+        message: 'resource match fails when current route does not match full url path',
+      },
+      {
+        currentRoute: {
+          url: '/posts/show/comments',
+          name: 'posts.show.comments',
+        },
+        routeValue: {
+          url: '/comments',
+          name: 'comments',
+        },
+        message: 'resource match fails when current route does not match the root route',
+      },
```

Comment:
@rwwagner90 You are very right. I did run into this edge case. The above test two cases roughly model this scenario.

**Edge case Failure**
![example-fail](https://user-images.githubusercontent.com/4172262/44797993-1153df00-ab76-11e8-8c99-cf0904953a63.gif)

Adding the following URL check `const hasMatchingURL = routeURL === currentRouteURL;` resolved this edge case.

**Edge Case Success**
![example-success](https://user-images.githubusercontent.com/4172262/44798057-35afbb80-ab76-11e8-84de-65b3008d042e.gif)

Please let me know if I failed to address your concern.